addons/tls13-rollout-bug1442042/bootstrap.js

```diff
@@ -0,0 +1,61 @@
+/* -*- indent-tabs-mode: nil; js-indent-level: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+let {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+
+Cu.import("resource://gre/modules/Preferences.jsm");
+Cu.import("resource://gre/modules/ClientID.jsm");
+
+Cu.importGlobalProperties(["crypto", "TextEncoder"]);
+
+// initially roll out to 10%, we want to control this on the client rather than 
+// depending on server-side throttling, as throttling cannot account for any
+// other concurrent gradual roll-outs.
+const ENABLE_PROB = 0.9;
+const DEBUG = false;
+const VERSION_MAX_PREF = "security.tls.version.max";
+
+function debug(msg) {
+  if (DEBUG) {
+    console.log(`TLS 1.3 Test: ${msg}`); // eslint-disable-line no-console
+  }
+}
+
+async function getRandomnessSeed() {
+  return ClientID.getClientID();
+}
+
+async function generateVariate(seed, label) {
+  const hasher = crypto.subtle;
+  const hash = await hasher.digest("SHA-256", new TextEncoder("utf-8").encode(seed + label));
+  let view = new DataView(hash);
+  return view.getUint32(0) / 0xffffffff;
+}
+
+async function startup(data, reason) {
+  // Don't do anything if the user has already messed with this
+  // setting.
+  let userprefs = new Preferences();
+  if (userprefs.isSet(VERSION_MAX_PREF)) {
+    debug("User has changed TLS max version. Skipping");
+    return;
+  }
+
+  debug("Installing");
+
+  let variate = await generateVariate(await getRandomnessSeed(), data.id);
```

Comment:
Is there any point passing in `data.id` here?  why not just hash the client id?

Reply:
Hey @ekr did you have a reason for this ^?

Reply:
This way if you run multiple experiments based on this template they independently select their cohorts rather than having the same people get every experiment